Makefile

```diff
@@ -1,36 +1,106 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This Makefile uses syntax that needs at least GNU Make version 3.82.
+# Test it & fail early. (MacOS still ships version 3.81 as of MacOS 15.)
+ifeq ($(filter undefine,$(value .FEATURES)),)
+$(error Unsupported version of Make. \
+    This Makefile does not work properly with GNU Make $(MAKE_VERSION); \
+    it needs GNU Make version 3.82 or later)
```

Comment:
Is this added to enable the ONESHELL below?

Consider rewriting build commands so they run in a standard shell-per-command invocation (adding end-of-line backslash for multi-line commands).

Reply:
> Is this added to enable the ONESHELL below?

I mainly wanted to make sure it's GNU make and not a different flavor of make. The 3.82 was because I reused these lines from what I did for some other Makefiles, and couldn't remember exactly the differences between 3.82 and previous versions. It saved time to just copy what I knew worked. The .ONESHELL behavior is just a benefit but not essential.

I am not sure if gnu make is the default on Windows.

Let's try removing this test and changing the shell commands to not require ONESHELL.

Reply:
OK, this is done.